miner/worker.go

```diff
@@ -1049,24 +1058,150 @@ func (w *worker) commitWork(interruptCh chan int32, timestamp int64) {
 		}
 		coinbase = w.coinbase // Use the preset address as the fee recipient
 	}
-	work, err := w.prepareWork(&generateParams{
-		timestamp: uint64(timestamp),
-		coinbase:  coinbase,
-	})
-	if err != nil {
-		return
-	}
 
-	// Fill pending transactions from the txpool
-	w.fillTransactions(interruptCh, work)
-	w.commit(work, w.fullTaskHook, true, start)
+	stopTimer := time.NewTimer(0)
+	defer stopTimer.Stop()
+	<-stopTimer.C // discard the initial tick
+
+	stopWaitTimer := time.NewTimer(0)
+	defer stopWaitTimer.Stop()
+	<-stopWaitTimer.C // discard the initial tick
+
+	// validator can try several times to get the most profitable block,
+	// as long as the timestamp is not reached.
+	workList := make([]*environment, 0, 10)
+	var bestWork *environment
+	// workList clean up
+	defer func() {
+		for _, w := range workList {
+			// only keep the best work, discard others.
+			if w == bestWork {
+				continue
+			}
+			w.discard()
```

Comment:
`w.discard()` will stop prefetcher, I think we should stop prefetcher once the mining is done for each block, no need to wait.

Reply:
In this way, we don't have to keep `workList`

Reply:
yes, we can remove the `workList`, but it may be better not to discard the bestWork. `TriePrefetch` may not be finished, better to let it keep working, so it will be faster to do block finalize.